### ISSUE TITLE
S1186: FP: Conditional compilation in method should not be considered empty

### ISSUE THREAD

**Issue by martin-strecker-sonarsource** [SonarSource/sonar-dotnet#5856]
### Description

see https://community.sonarsource.com/t/csharpsquid-s1186-methods-should-not-be-empty-false-positive/67093

"S1186 Methods should not be empty" should not be raised here:

```cs
public void JobAdded(IJobDetail jobDetail) {
#if DEBUG
    _logManager.Log(LogType.Database | LogType.File, new LogMessage {
            MessageType = LogMessageType.Info,
            CustomData = { { "JobAdded", jobDetail.JobType.Name } },
            MessageCategory = this.GetTypeName()
    });
#endif
}
```
